### PR TITLE
New Rule: (disallow|require)PaddingNewLinesBeforeExport

### LINKED ISSUE #1220: New Rule: Blank line after 'use strict', New Rule: Blank line before module.exports
I've been working on a style guide for CommonJS specific things and it has brought to light a few things that would be good JSCS rules.

The style guide:
https://gist.github.com/TheSavior/af94986da0ecef32e310

The examples:

## Leave a blank line after strict mode declaration
**Good**
```javascript
'use strict';

// code
```

**Bad**
```javascript
'use strict';
// code
```

## Always put a blank line before your export statement
**Good**
```javascript
// code

module.exports = {};
```

**Bad**
```javascript
// code
module.exports = {};
```

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/10237536-new-rule-blank-line-after-use-strict-new-rule-blank-line-before-module-exports?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by TheSavior**
As discussed in #1220 

Closes #1220.

### CODE REVIEW

File: lib/rules/require-padding-newlines-before-export.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+    configure: function(value) {
+        assert(
+            value === true,
+            this.getOptionName() + ' option requires a true value or should be removed'
+        );
+    },
+
+    getOptionName: function() {
+        return 'requirePaddingNewLinesBeforeExport';
```
Comment (zxqfox) [inline lib/rules/require-padding-newlines-before-export.js]: `NewLines` or `Newlines` or `Newline`?

Comment (TheSavior) [inline lib/rules/require-padding-newlines-before-export.js]: Yeah, seems weird to me too. However, it matches our other rules:

disallowPaddingNewLinesAfterBlocks

Comment (zxqfox) [inline lib/rules/require-padding-newlines-before-export.js]: But: https://github.com/jscs-dev/node-jscs/pull/798#issuecomment-65096090
And disallowNewlineBeforeBlockStatements, disallowPaddingNewlinesBeforeKeywords, `-newlines-`

I think it was accidentally names NewLines (with a capital L) and then just copy-pasted several times ;-(

Comment (TheSavior) [inline lib/rules/require-padding-newlines-before-export.js]: Fixing these rules should be done in a separate PR then and will likely need to deprecate these ones so we don't break backwards compatibility, right?

Comment (zxqfox) [inline lib/rules/require-padding-newlines-before-export.js]: Yeah, I think so. For v2.0 ;-)
